### PR TITLE
plumbing for publication of synapse 3.0.0 alpha wheels to vtx pypi services.

### PR DESCRIPTION
**Pull request by vEpiphyte**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -252,6 +252,85 @@ commands:
...
             twine upload dist/*
 
+  deploy_to_devpi:
+    description: "Publish wheel packages to devpi"
+    parameters:
+      devpi_base_url:
+        description: Base URL for the devpi service
+        type: string
+      devpi_index:
+        description: Index to push packages too and install requirements from.
```
Comment (Cisphyx) [inline .circleci/config.yml:262]:
```suggestion
        description: Index to push packages to and install requirements from.
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -528,6 +607,26 @@ jobs:
...
+      - image: cimg/python:3.11
+    steps:
+      - checkout
+      - run:
+          name: Version bump and date the build.
+          command: |
+            python3 -m pip install bump2version
+            DATE=`date '+%Y%m%d%H%M'`
+            CURRENT_VERSION=`grep "current_version =" .bumpversion.cfg | cut -f 2 -d "=" | tr -d '[:space:]'`
+            # The "patch" value here just to fulfill a required argument.
```
Comment (Cisphyx) [inline .circleci/config.yml:621]:
```suggestion
            # The "patch" value is used here just to fulfill a required argument.
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -528,6 +607,26 @@ jobs:
...
+    docker:
+      - image: cimg/python:3.11
+    steps:
+      - checkout
+      - run:
+          name: Version bump and date the build.
+          command: |
+            python3 -m pip install bump2version
+            DATE=`date '+%Y%m%d%H%M'`
+            CURRENT_VERSION=`grep "current_version =" .bumpversion.cfg | cut -f 2 -d "=" | tr -d '[:space:]'`
```
Comment (Cisphyx) [inline .circleci/config.yml:620]: I think this also needs to strip off the previous `a{DATE}` portion of the version

Comment (vEpiphyte) [reply .circleci/config.yml:620]: We're not persisting these values in git, so we shouldn't need to strip anything here. They are only being used to publish the wheels to the internal index.

Comment (Cisphyx) [reply .circleci/config.yml:620]: Oh I see, gotcha

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**